src/flashbar/__tests__/collapsible.test.tsx

```diff
@@ -0,0 +1,323 @@
+      expect(expandedItems[0].findContent()!.getElement()).toHaveTextContent('Everything went fine');
+      expect(expandedItems[1].findHeader()!.getElement()).toHaveTextContent('Error');
+      expect(expandedItems[1].findContent()!.getElement()).toHaveTextContent('There was an error');
+
+      findOuterToggleElement(flashbar)!.click();
+
+      const collapsedItems = flashbar.findItems();
+      expect(collapsedItems.length).toBe(1);
+      expect(collapsedItems[0].findHeader()!.getElement()).toHaveTextContent('Success');
+      expect(collapsedItems[0].findContent()!.getElement()).toHaveTextContent('Everything went fine');
+    });
+
+    it('collapses automatically when enough items are added', () => {
+      const item1 = { ...sampleItems.success, id: '0' };
+      const item2 = { ...sampleItems.error, id: '1' };
+
+      const { container, rerender } = render(
+        <Flashbar items={[item1]} {...{ collapsible: true, i18nStrings: defaultStrings }} />
+      );
+      const wrapper = createWrapper(container);
+      const flashbar = wrapper.findFlashbar()!;
+      expect(flashbar.findItems()).toHaveLength(1);
+      expect(findOuterToggleElement(flashbar)).toBeFalsy();
+
+      rerender(<Flashbar items={[item1, item2]} {...{ collapsible: true, i18nStrings: defaultStrings }} />);
+      expect(wrapper.findFlashbar()!.findItems()).toHaveLength(1);
+      const toggleElement = findOuterToggleElement(wrapper.findFlashbar()!);
+      expect(toggleElement).toBeTruthy();
+    });
+
+    it('collapses automatically again when enough items are added even if it had been expanded before', () => {
+      const item1 = { ...sampleItems.success, id: '0' };
+      const item2 = { ...sampleItems.error, id: '1' };
+
+      const { container, rerender } = render(
+        <Flashbar items={[item1, item2]} {...{ collapsible: true, i18nStrings: defaultStrings }} />
+      );
+      const wrapper = createWrapper(container);
+      const flashbar = wrapper.findFlashbar()!;
+      expect(flashbar.findItems()).toHaveLength(1);
+      expect(findOuterToggleElement(flashbar)).toBeTruthy();
+      const toggleElement = findOuterToggleElement(wrapper.findFlashbar()!);
+      expect(toggleElement).toBeTruthy();
+      toggleElement!.click();
+      expect(flashbar.findItems()).toHaveLength(2);
+
+      rerender(<Flashbar items={[item1]} {...{ collapsible: true, i18nStrings: defaultStrings }} />);
+      expect(wrapper.findFlashbar()!.findItems()).toHaveLength(1);
+      expect(findOuterToggleElement(wrapper.findFlashbar()!)).toBeFalsy();
+
+      rerender(<Flashbar items={[item1, item2]} {...{ collapsible: true, i18nStrings: defaultStrings }} />);
+      expect(wrapper.findFlashbar()!.findItems()).toHaveLength(1);
+      expect(findOuterToggleElement(wrapper.findFlashbar()!)).toBeTruthy();
+    });
+  });
+
+  describe('Accessibility', () => {
+    it('renders items in an unordered list', () => {
```

Comment:
What is meant is that it is rendered as an UL element (versus ordered list, OL). The order of the items however is kept of course, and this seems to work as desired with screen readers. This was also an explicit a11y requirement. Also, the non-collapsible Flashbar also uses UL, so I think if we want to reconsider it, we should do it also for the non-collapsible Flashbar, which means: 1) IMHO it would be better to do it separately; 2) I can imagine this breaking unexpected things?

Reply:
Yeah, this is completely outside of the scope for this PR. It just got me wondering why we don't use `ol` in general for flashbar.